### PR TITLE
fix(flex): Fix hqps engine's support for some simple `MATCH` queries 

### LINKED ISSUE #3080: [BUG] High QPS Engine in Flex currently lacks support for MultiLabel vertex
High QPS Engine fails to run cypher query like 
```cypher
MATCH(n) RETURN n;
MATCH(n)  RETURN n LIMIT 1;
MATCH(n  { name : "alice" }) RETURN N;
```

### PR DESCRIPTION
**Pull request by zhanglei1949**
## What do these changes do?

- Add a test suite containing 9 simple match queries for hqps Cypher CI test.
- Refactor the implementation of data structure like `GeneralVertexSet`, `FlatEdgeSet`.
- Fix  the implementation of graph operators for multiLabel VertexSet and EdgeSet.

## Related issue number

Fixes #3080

### CODE REVIEW

**Review by luoxiaojian — APPROVED** [review]

File: flex/codegen/src/hqps/hqps_scan_builder.h [hunk 1]
```diff
@@ -79,7 +83,7 @@ class ScanOpBuilder {
   // get required oid from query params
   ScanOpBuilder& queryParams(const algebra::QueryParams& query_params) {
     if (!query_params.has_predicate()) {
-      throw std::runtime_error(std::string("expect expr in params"));
+      LOG(WARNING) << "No expr in params";
```
Comment (longbinlai) [inline flex/codegen/src/hqps/hqps_scan_builder.h:86]: The warning log is not necessary as it is common to have no parameter

**Comment by zhanglei1949** [comment]
Still working on this...

File: flex/codegen/src/hqps/hqps_scan_builder.h (hunk 1 above)
Comment (zhanglei1949) [reply flex/codegen/src/hqps/hqps_scan_builder.h:86]: Use `VLOG(10)` instead.

**Review by longbinlai — APPROVED** [review]
LGTM

**Review by luoxiaojian — APPROVED** [review]

**Pull request merged**